FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_DNS.c

```diff
@@ -1072,39 +1447,86 @@ when ipconfigDNS_USE_CALLBACKS == 1
 when ipconfigUSE_LLMNR == 1
 for testing purposes, by the module iot_test_freertos_tcp.c
 */
-uint32_t ulDNSHandlePacket( const NetworkBufferDescriptor_t *pxNetworkBuffer )
+extern void show_addressinfo( struct freertos_addrinfo *pxAddress );
+
+void show_addressinfo( struct freertos_addrinfo *pxAddress )
 {
-DNSMessage_t *pxDNSMessageHeader;
-size_t uxPayloadSize;
+	struct freertos_addrinfo *ptr = pxAddress;
+	BaseType_t xIndex = 0;
+	while( ptr != NULL )
+	{
+	char cBuffer[ 40 ];
+	const uint8_t *pucAddress;
+
+	#if( ipconfigUSE_IPv6 != 0 )
+		if( ptr->ai_family == FREERTOS_AF_INET6 )
+		{
+		struct freertos_sockaddr6 *sockaddr6 = ( struct freertos_sockaddr6 * ) ptr->ai_addr;
+
+			pucAddress = ( const uint8_t * ) &( sockaddr6->sin_addrv6 );
+		}
+		else
+	#endif /* ( ipconfigUSE_IPv6 != 0 ) */
+		{
+			pucAddress = ( const uint8_t * ) &( ptr->ai_addr->sin_addr );
+		}
+		FreeRTOS_inet_ntop( ptr->ai_family, ( const void * ) pucAddress, cBuffer, sizeof( cBuffer ) );
+		FreeRTOS_printf( ( "Address %d: %s\n", ( int ) ++xIndex, cBuffer ) );
+		ptr = ptr->ai_next;
+	}
+}
 
-	/* Only proceed if the payload length indicated in the header
-	appears to be valid. */
-	if( pxNetworkBuffer->xDataLength >= sizeof( UDPPacket_t ) )
+
+/* The function below will only be called :
+when ipconfigDNS_USE_CALLBACKS == 1
+when ipconfigUSE_LLMNR == 1
+for testing purposes, by the module iot_test_freertos_tcp.c
+*/
+#if( ipconfigUSE_DNS == 1 ) && ( ( ipconfigDNS_USE_CALLBACKS == 1 ) || ( ipconfigUSE_LLMNR == 1 ) )
+
+	uint32_t ulDNSHandlePacket( const NetworkBufferDescriptor_t *pxNetworkBuffer )
 	{
-		uxPayloadSize = pxNetworkBuffer->xDataLength - sizeof( UDPPacket_t );
+	DNSMessage_t *pxDNSMessageHeader;
+	size_t uxPayloadSize;
 
-		if( uxPayloadSize >= sizeof( DNSMessage_t ) )
+		/* Only proceed if the payload length indicated in the header
+		appears to be valid. */
+		if( pxNetworkBuffer->xDataLength >= sizeof( UDPPacket_t ) )
 		{
-			pxDNSMessageHeader =
-				ipCAST_PTR_TO_TYPE_PTR( DNSMessage_t, pxNetworkBuffer->pucEthernetBuffer );
+			uxPayloadSize = pxNetworkBuffer->xDataLength - sizeof( UDPPacket_t );
 
-			/* The parameter pdFALSE indicates that the reply was not expected. */
-			( void ) prvParseDNSReply( ( uint8_t * ) pxDNSMessageHeader,
-				uxPayloadSize,
-				pdFALSE );
+			if( uxPayloadSize >= sizeof( DNSMessage_t ) )
+			{
+			struct freertos_addrinfo *pxAddressInfo = NULL;
+
+				pxDNSMessageHeader =
+					ipPOINTER_CAST( DNSMessage_t *, &( pxNetworkBuffer->pucEthernetBuffer [ sizeof( UDPPacket_t ) ] ) );
```

Comment:
Instance.

Reply:
I did all instanced except 4, that are casting to standard types. Not sure about them.